controllers/openstackcluster_controller.go

```diff
@@ -503,44 +509,50 @@ func (r *OpenStackClusterReconciler) SetupWithManager(mgr ctrl.Manager, options
 				return false
 			},
 			DeleteFunc: func(e event.DeleteEvent) bool {
-				log := r.Log.WithValues("predicate", "deleteEvent", "namespace", e.Meta.GetNamespace(), "cluster", e.Meta.GetName())
+				log := log.WithValues("predicate", "deleteEvent", "namespace", e.Object.GetNamespace(), "cluster", e.Object.GetName())
 				log.V(4).Info("Cluster did not match expected conditions, will not attempt to map associated OpenStackCluster.")
 				return false
 			},
 			GenericFunc: func(e event.GenericEvent) bool {
-				log := r.Log.WithValues("predicate", "genericEvent", "namespace", e.Meta.GetNamespace(), "cluster", e.Meta.GetName())
+				log := log.WithValues("predicate", "genericEvent", "namespace", e.Object.GetNamespace(), "cluster", e.Object.GetName())
 				log.V(4).Info("Cluster did not match expected conditions, will not attempt to map associated OpenStackCluster.")
 				return false
 			},
 		},
 	)
 }
 
-func (r *OpenStackClusterReconciler) requeueOpenStackClusterForUnpausedCluster(o handler.MapObject) []ctrl.Request {
-	c := o.Object.(*clusterv1.Cluster)
-	log := r.Log.WithValues("objectMapper", "clusterToOpenStackCluster", "namespace", c.Namespace, "cluster", c.Name)
+func (r *OpenStackClusterReconciler) requeueOpenStackClusterForUnpausedCluster(log logr.Logger) handler.MapFunc {
```

Comment:
This method can be removed in favor of using `util.ClusterToInfrastructureMapFunc(infrav1.GroupVersion.WithKind("OpenStackCluster")`

Reply:
Done